#! /bin/sh /usr/share/dpatch/dpatch-run
## 01_uml_net - Mattia Dongili <malattia@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Use Debian's standard location for uml_net helper

@DPATCH@

# Author: malattia
# Status: none

--- arch/um/os-Linux/helper.c~clean	2006-04-02 16:32:38.340801686 +0200
+++ arch/um/os-Linux/helper.c	2006-04-02 16:34:13.454745936 +0200
@@ -22,6 +22,9 @@ struct helper_data {
 	int fd;
 };
 
+/* PATH variable to find uml helpers  installed in FHS compliant locations */
+static char *alt_path = "PATH=:/bin:/usr/bin:/usr/lib/uml"; 
+
 /* Debugging aid, changed only from gdb */
 int helper_pause = 0;
 
@@ -41,6 +44,7 @@ static int helper_child(void *arg)
 	}
 	if(data->pre_exec != NULL)
 		(*data->pre_exec)(data->pre_data);
+	putenv(alt_path);
 	execvp(argv[0], argv);
 	errval = errno;
 	printk("execvp of '%s' failed - errno = %d\n", argv[0], errno);
